sql/main_test.go

```diff
@@ -39,8 +43,39 @@ func TestMain(m *testing.M) {
 	leaktest.TestMainWithLeakCheck(m)
 }
 
+// checkEndTransactionTrigger verifies that an EndTransactionRequest
+// that includes intents for the SystemDB keys sets the proper trigger.
+func checkEndTransactionTrigger(req proto.Request) error {
+	args, ok := req.(*proto.EndTransactionRequest)
+	if !ok {
+		return nil
+	}
+
+	if !args.Commit {
+		// This is a rollback: skip trigger verification.
+		return nil
+	}
+
+	hasTrigger := args.GetInternalCommitTrigger().GetModifiedSpanTrigger().GetSystemDBSpan()
+
+	for _, it := range args.Intents {
+		if keys.SystemDBSpan.ContainsKey(it.GetKey()) {
+			if !hasTrigger {
+				return util.Errorf("EndTransaction contains key %s, but is missing SystemDBSpan trigger", it.GetKey())
```

Comment:
should this also check the inverse? i.e. that the trigger is not set unnecessarily